docs/calendar-draft.md

```diff
@@ -0,0 +1,327 @@
+
+## Temporal.Date internal slots
+
+Temporal.Date currently has three internal slots: year, month, and day. (An "internal slot" refers to actual data, as opposed to "properties", which could be computed.)  In this proposal, three more internal slots would be added:
+
+1. calendar
+2. isLeapMonth
+3. era
+
+It is believed that a date in any known calendar can be described using the five-tuple (*year*, *month*, *day*, *isLeapMonth*, *era*).  Since all operations involving the slots of Temporal.Date will go through the calendar first, calendars that need more data can assign complex types to these slots.  It is possible that *isLeapMonth* can be merged into *month* by expanding the set of numbers that *month* can represent.
+
+The *calendar* slot contains an object implementing the Temporal.Calendar interface, described below.
+
+## Temporal.Calendar interface
+
+The new Temporal.Calendar interface is a mechanism to allow arbitrary calendar systems to be implemented on top of Temporal.  ***Most users will not encounter the Temporal.Calendar interface directly***, unless they are building or using a non-built-in calendar system.
+
+An open question is whether this "interface" should be a protocol or an identity.  See issue #289.  The following section assumes it is a protocol.
+
+### Methods on the Temporal.Calendar interface
+
+All of the following methods return new Temporal objects.  All properties of `Temporal.Calendar` are Symbols.
+
+```javascript
+class MyCalendar {
+
+	///////////////////
+	//  To/From ISO  //
+	///////////////////
+
+	/** Returns the projection of self in the ISO calendar */
+	[Temporal.Calendar.toISO](
+		self: Temporal.Date
+	) : Temporal.Date;
+
+	/** Returns the projection of isoDate in the custom calendar */
+	[Temporal.Calendar.fromISO](
+		isoDate: Temporal.Date
+	) : Temporal.Date;
+
+	/** A string identifier for this calendar */
+	[Temporal.Calendar.id] : string;
+
+	//////////////////
+	//  Arithmetic  //
+	//////////////////
+
+	/** Returns self plus duration according to the calendar rules. */
+	[Temporal.Calendar.plus](
+		self: Temporal.Date,
+		duration: Temporal.Duration,
+		options: /* options bag */
+	) : Temporal.Date;
+
+	/** Returns self minus duration according to the calendar rules. */
+	[Temporal.Calendar.minus](
+		self: Temporal.Date,
+		duration: Temporal.Duration,
+		options: /* options bag */
+	) : Temporal.Date;
+
+	/** Returns self minus other, which are dates in the same calendar. */
+	[Temporal.Calendar.difference](
+		self: Temporal.Date,
+		other: Temporal.Date,
+		options: /* options bag */
+	) : Temporal.Duration;
+
+	////////////////////////////////////
+	//  Accessors:                    //
+	//  Semantics defined in date.md  //
+	////////////////////////////////////
+
+	[Temporal.Calendar.dayOfWeek](
```

Comment:
would all these symbols be the same across realms, like normal well-known symbols?

Reply:
Copied to #289